java/src/main/java/com/google/appengine/tools/pipeline/impl/servlets/PipelineServlet.java

```diff
@@ -88,6 +91,28 @@ public boolean matches(String path) {
     return Pair.of(path, requestType);
   }
 
+  transient AbortJobHandler abortJobHandler;
+  transient DeleteJobHandler deleteJobHandler;
+  transient JsonClassFilterHandler jsonClassFilterHandler;
+  transient JsonListHandler jsonListHandler;
+  transient JsonTreeHandler jsonTreeHandler;
+  transient TaskHandler taskHandler;
+
+  @SneakyThrows
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    //TODO: coupled to GAE; could we pull this via ServletConfig somehow, to make more flexible?
+    PipelineManager pipelineManager =
+      new PipelineManager(SystemProperty.applicationId.get(), GoogleCredentials.getApplicationDefault());
+    abortJobHandler = new AbortJobHandler(pipelineManager);
+    deleteJobHandler = new DeleteJobHandler(pipelineManager);
+    jsonClassFilterHandler = new JsonClassFilterHandler(pipelineManager);
+    jsonListHandler = new JsonListHandler(pipelineManager);
+    jsonTreeHandler = new JsonTreeHandler(pipelineManager);
+    taskHandler = new TaskHandler(pipelineManager);
```

Comment:
servlet is the controller; then these are all the sub-controllers; better way to handle this?

Reply:
ugly as hell, but it sort of ok. this is the front controller, yes, so at least responsibilities are well split.